pvlib/inverter.py

```diff
@@ -91,29 +124,80 @@ def sandia(v_dc, p_dc, inverter):
     '''
 
     Paco = inverter['Paco']
-    Pdco = inverter['Pdco']
-    Vdco = inverter['Vdco']
-    Pso = inverter['Pso']
-    C0 = inverter['C0']
-    C1 = inverter['C1']
-    C2 = inverter['C2']
-    C3 = inverter['C3']
     Pnt = inverter['Pnt']
+    Pso = inverter['Pso']
 
-    A = Pdco * (1 + C1 * (v_dc - Vdco))
-    B = Pso * (1 + C2 * (v_dc - Vdco))
-    C = C0 * (1 + C3 * (v_dc - Vdco))
-
-    power_ac = (Paco / (A - B) - C * (A - B)) * (p_dc - B) + C * (p_dc - B)**2
-    power_ac = np.minimum(Paco, power_ac)
-    power_ac = np.where(p_dc < Pso, -1.0 * abs(Pnt), power_ac)
+    power_ac = _sandia_eff(v_dc, p_dc, inverter)
+    power_ac = _sandia_limits(power_ac, p_dc, Paco, Pnt, Pso)
 
     if isinstance(p_dc, pd.Series):
         power_ac = pd.Series(power_ac, index=p_dc.index)
 
     return power_ac
 
 
+def sandia_multi(v_dc, p_dc, inverter):
+    r'''
+    Convert DC power and voltage to AC power for an inverter with multiple
+    MPPT inputs.
+
+    Uses Sandia's Grid-Connected PV Inverter model [1]_.
+
+    Parameters
+    ----------
+    v_dc : numeric or tuple of numeric
+        DC voltage on each MPPT input of the inverter. [V]
+
+    p_dc : numeric or tuple of numeric
+        DC power on each MPPT input of the inverter. [W]
+
+    inverter : dict-like
+        Defines parameters for the inverter model in [1]_.
+
+    Returns
+    -------
+    power_ac : numeric
+        AC power output for the inverter. [W]
+
+    Raises
+    ------
+    ValueError if v_dc and p_dc have different lengths.
+
+    Notes
+    -----
+    See :py:func:`pvlib.inverter.sandia` for definition of the parameters in
+    `inverter`.
+
+    References
+    ----------
+    .. [1] D. King, S. Gonzalez, G. Galbraith, W. Boyson, "Performance Model
+       for Grid-Connected Photovoltaic Inverters", SAND2007-5036, Sandia
+       National Laboratories.
+
+    See also
+    --------
+    pvlib.inverter.sandia
+    '''
+
+    if isinstance(p_dc, tuple):
+        if isinstance(v_dc, tuple) and not len(p_dc) == len(v_dc):
+            raise ValueError('p_dc and v_dc have different lengths')
+        power_dc = sum(p_dc)
+    else:
+        power_dc = sum((p_dc,))  # handle Series, array or float
+
+    power_ac = 0. * power_dc
+
+    if isinstance(p_dc, tuple):
+        for vdc, pdc in zip(v_dc, p_dc):
+            power_ac += pdc / power_dc * _sandia_eff(vdc, power_dc, inverter)
+    else:
+        power_ac = _sandia_eff(v_dc, power_dc, inverter)
```

Comment:
I don't understand why we'd include this line in the `sandia_multi` function. Isn't it duplicating functionality in `sandia`? Maybe I don't understand the use cases well enough.

Reply:
I was thinking that `inverter.sandia_multi(600., 1500., *params)` should work, but can't really say that's a required use.

Eventually I'd see `inverter.sandia` and `inverter.sandia_multi` merging, once the latter is better documented, with validation, and has broader acceptance.